pkg/proofs/ethereum/proof.go

```diff
@@ -160,18 +159,18 @@ func NewTrie(list types.DerivableList) Trie {
 		// #nosec G115 iterator
 		indexBuf = rlp.AppendUint64(indexBuf[:0], uint64(i))
 		value := encodeForDerive(list, i, valueBuf)
-		hasher.Update(indexBuf, value)
+		_ = hasher.Update(indexBuf, value)
```

Comment:
Consider logging the error , even if we don't need to handle it

Reply:
There is no logger here. The error condition is not reachable in this case. I think the least bad thing to do if we want to handle this is to just panic (since it really should never happen)